common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -51,3 +54,105 @@ def perform_destroy(self, instance):
             )
         if save_model:
             instance.save()
+
+
+class EntitlementEnrollmentViewSet(viewsets.GenericViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = CourseEntitlementSerializer
+
+    def _enroll_entitlement(self, entitlement, course_session_key, user):
+        enrollment = CourseEnrollment.enroll(
+            user=user,
+            course_key=course_session_key,
+            mode=entitlement.mode,
+        )
+
+        CourseEntitlement.set_enrollment(entitlement, enrollment)
+
+    def _unenroll_entitlement(self, entitlement, course_session_key, user):
+        CourseEnrollment.unenroll(user, course_session_key, skip_refund=True)
+        CourseEntitlement.set_enrollment(entitlement, None)
+
+    def create(self, request, uuid):
+        course_session_id = request.data.get('course_run_id', None)
+
+        if not course_session_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Course Run ID was not provided."
+            )
+
+        # Verify that the user has an Entitlement for the provided Course UUID.
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=uuid, user=request.user, expired_at=None)
+        except CourseEntitlement.DoesNotExist:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Entitlement for this UUID does not exist or is Expired."
+            )
+
+        # Verify the course run ID is of the same type as the Course entitlement.
+        course_run_valid = False
+        course_runs = get_course_runs_for_course(entitlement.course_uuid)
+        for run in course_runs:
+            if course_session_id == run.get('key', ''):
```

Comment:
Do we need to normalize the types before we make this comparison? (`str` vs `unicode`)?